packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -541,6 +559,15 @@ export class DateTimePicker extends ManageHelpText(
         return html`
             <div
                 role="spinbutton"
+                aria-valuenow=${ifDefined(segment.value)}
+                aria-valuemin=${segment.minValue}
+                aria-valuemax=${segment.maxValue}
+                aria-label=${segment.label}
+                aria-valuetext=${ifDefined(
+                    segment.value !== undefined
+                        ? segment.formatted
+                        : this.labels.empty
+                )}
```

Comment:
With this approach it seems like on Chrome it doesn't always read the whole value of the segment but instead the VoiceOver announces "selection replaced {number}". Also, deleting the segment's value makes the VO announce the content of the segment (----) instead of "Empty".
Also, on Safari the VO announces "beginning of text" instead of the segment's value when they change using the ArrowUp/Down, and when trying to use the VO + ArrowUp/Down to change the stepper's value it ocasionally crashed the webpage.

I'm currently exploring ways of fixing these issues. In the meantime, could someone please suggest some ideas? Thank you!

Reply:
@nikkimk ^